### PR TITLE
Add optional tags field to objectives and indicators

### PR DESCRIPTION
**Pull request by weyert**
Introduces the optional field `tags` to allow to associate tags with objectives or indicators. Allowing systems that leverage the OpensLO files to display tags, or allow easy filtering by tags in UI interfaces

I am not 100% sure if it makes sense to add the field to both of it, I think there was some discussion about potentially moving out the objectives into a separate file. Happy to discuss it.

An example would be:

```
apiVersion: openslo/v1alpha
kind: SLO
metadata:
  name: slo-my-service
  displayName: Requests Availability
spec:
  service: my-service
  tags: 
     - availability
     - common-slo
     - my-service
  description: "Common SLO based on availability for HTTP request responses."
  budgetingMethod: Occurrences
  objectives:
    - ratioMetrics:
        good:
          source: prometheus
          queryType: promql
          query: sum(rate(http_request_duration_seconds_count{job="myservice",code!~"(5..|429)"}[{{.window}}]))
        total:
          source: prometheus
          queryType: promql
          query: sum(rate(http_request_duration_seconds_count{job="myservice"}[{{.window}}]))
      target: 0.97
  timeWindows:
    - count: 30
       unit: Day
```

### CODE REVIEW

**Comment by weyert** [comment]
I am not sure if it would be better to be define it as an optional field on the `metadata`-blob? 
E.g.
```yaml
apiVersion: openslo/v1alpha
kind: SLO
metadata:
  name: slo-my-service
  displayName: Requests Availability
  tags: 
     - availability
     - common-slo
     - my-service
spec:
  service: my-service
  description: "Common SLO based on availability for HTTP request responses."
  budgetingMethod: Occurrences
  objectives:
    - ratioMetrics:
        good:
          source: prometheus
          queryType: promql
          query: sum(rate(http_request_duration_seconds_count{job="myservice",code!~"(5..|429)"}[{{.window}}]))
        total:
          source: prometheus
          queryType: promql
          query: sum(rate(http_request_duration_seconds_count{job="myservice"}[{{.window}}]))
      target: 0.97
  timeWindows:
    - count: 30
       unit: Day
```

**Comment by ian-bartholomew** [comment]
I think that it makes more sense in the `metadata` field FWIW

**Comment by geototti21** [comment]
This is nice! I created #57 to introduce `annotations`, should we do something similar here and introduce `labels` instead of `tags` to follow k8s patterns?

**Comment by ian-bartholomew** [comment]
Some of the use cases that i can see are:
- grouping SLOs 
- filtering SLOs
- searching for SLOs
- We could also have an alert policy that uses labels or tags as the criteria 

Personally i don't have a strong feeling about tags vs labels, but labels seems more k8s like

**Comment by ian-bartholomew** [comment]
Thinking about it more, i think`labels` might be better to use,  with the child elements as arrays, so that you can have multiple values for a given key.  Eg

```yaml
metadata:
  labels:                                                                                                                                                                                                                           
    bar:                                                                                                                                                                                                                            
    - bar                                                                                                                                                                                                                           
    - baz                                                                                                                                                                                                                           
    foo:                                                                                                                                                                                                                            
    - bar                                                                                                                                                                                                                           
    - baz    
```

**Comment by nicolastakashi** [comment]
The idea to have labels is awesome, as well as the annotation proposed by the @geototti21. We can use labels to group SLOs from different services, but that belongs to the same boundary.

**Comment by weyert** [comment]
Sounds like a good idea to make it more flexible. Yeah, probably specific tags might not needed if the entities have a `metadata`-blob. I think this would be mean their would be `metadata`-fields for:

- objectives
- indicators
- metrics query definitions (the `good`, `total` of `ratioMetrics`)

File: README.md [hunk 1]
```diff
@@ -72,6 +72,11 @@ kind: SLO
 metadata:
   name: string
   displayName: string # optional
+  labels: # optional, key <>value a pair of labels that can be used as metadata
```
Comment (geototti21) [inline README.md:75]:
```suggestion
  labels: # optional, key <>value a pair of labels that can be used as metadata relevant to users
```

File: README.md [hunk 2]
```diff
@@ -101,6 +106,11 @@ spec:
 
 ##### Notes (SLO)
 
+- **metadata.labels:** *map[string]string|string[]* - optional field `key` <> `value`
+  - the `name` segment is required and must contain at most 63 characters beginning and ending
```
Comment (geototti21) [inline README.md:110]:
```suggestion
  - the `key` segment is required and must contain at most 63 characters beginning and ending
```

File: README.md [hunk 3]
```diff
@@ -101,6 +106,11 @@ spec:
 
 ##### Notes (SLO)
 
+- **metadata.labels:** *map[string]string|string[]* - optional field `key` <> `value`
```
Comment (geototti21) [inline README.md:109]: Shall we add one point with one / two examples in here, as I added for `annotations` [here](https://github.com/OpenSLO/OpenSLO/pull/64/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R115) so its easier in the future people to understand what goes where 🙏

**Review by geototti21 — COMMENTED** [review]
Looks good!  left you some suggestions 🤓

File: README.md (hunk 3 above)
Comment (weyert) [reply README.md:111]: Great suggestions,  I have updated the spec with your suggestions and also added some example labels (`costCentre`, `userImpacting`, `team`)

Comment (weyert) [reply README.md:111]: @geototti21 Please double check :)

**Review by geototti21 — APPROVED** [review]
🍰 nice one

**Review by nobl9-mikec — APPROVED** [review]
awesome pairing w/ Annotations!

**Comment by nobl9-mikec** [comment]
@weyert sorry about the merge conflict from Annotations! Would you mind resolving and then I can merge this in?

**Comment by weyert** [comment]
@nobl9-mikec Cool, should be resolved now :)

**Pull request merged**